File: 프로그래머스/1/42840. 모의고사/모의고사.kt
class Solution {
    fun solution(answers: IntArray): IntArray {
        var count = intArrayOf(0,0,0) //count 저장
        var student:List<IntArray> = listOf(
            intArrayOf(1,2,3,4,5),
            intArrayOf(2,1,2,3,2,4,2,5),
            intArrayOf(3,3,1,1,2,2,4,4,5,5)
        )
        
        //각 자리에 매치해야됨.
        for((index, ele) in answers.withIndex()){
            //answers값이 끝날때까지 돌아, 
            //student값을 돌려
            student.forEachIndexed{ stdIdx, it ->
                if(ele == it[index%it.size])
                    count[stdIdx] ++
            }
        }
        
        var Max = count.maxOrNull()?: 0 //최고 값
        //Max랑 같은 값일 경우 모두 추출 
        
        return count.mapIndexed{index, value -> 
            if(value == Max){
                index+1
            }else null
        }.filterNotNull().toIntArray()
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    var stdAnswer = [];
    let count = [0,0,0];
    const student = [[1,2,3,4,5],
                     [2,1,2,3,2,4,2,5],
                    [3,3,1,1,2,2,4,4,5,5]
                    ];
    
    answers.forEach((answer, ansIdx)=>{
        //2중 배열인 student를 돔 
        student.forEach((std, stdIdx)=>{
            if(answer == std[ansIdx%std.length]){
                count[stdIdx]++
            }  
        })
    })
    //가장 큰 수의 인덱스
    let stdMax = Math.max(...count)
    count.forEach((value, index)=>{
        if(value == stdMax){
            stdAnswer.push(index+1)
        }              
    })
    
    return stdAnswer;
}